Source/SourceKittenFramework/String+SourceKitten.swift

```diff
@@ -21,7 +21,66 @@ private let commentLinePrefixCharacterSet: NSCharacterSet = {
   return characterSet
 }()
 
+private var keyByteOffsetCache = "ByteOffsetCache"
+
 extension NSString {
+    /**
+    ByteOffsetCache is cache byte offset and UTF8Index of UTF16 based location.
+    This depends on using for that will be calculated from begening of the string to the end.
+    */
+    @objc private class ByteOffsetCache: NSObject {
+        struct Entry {
+            let byteOffset: Int
+            let index: String.UTF8Index
+        }
+        
+        var cache = Dictionary<Int, Entry>()
+        let utf8View: String.UTF8View
+        
+        init(_ string: String) {
+            self.utf8View = string.utf8
+        }
+        
+        func byteOffsetFromLocation(location: Int, andIndex index: String.UTF8Index) -> Int {
+            if let entry = cache[location] {
+                return entry.byteOffset
+            } else {
+                let entry: Entry
+                if let nearestLocation = cache.keys.filter({ $0 < location }).maxElement() {
+                    let nearestEntry = cache[nearestLocation]!
+                    let byteOffset = nearestEntry.byteOffset + nearestEntry.index.distanceTo(index)
+                    entry = Entry(byteOffset: byteOffset, index: index)
+                } else {
+                    let byteOffset = utf8View.startIndex.distanceTo(index)
+                    entry = Entry(byteOffset: byteOffset, index: index)
+                }
+                cache[location] = entry
+                
+                // keep largest 5 entry
+                if cache.keys.count > 5 {
```

Comment:
Now I tested again.
I confirmed that evicting cache does not affect performance. :+1: 
I'll remove that.